### PR TITLE
WIP: Updated tabs, Items and created a new features for tabsblock, tabItemIcon, TabView components.

### PR DESCRIPTION
**Pull request by cfmitrah**
Updated new features in tabs component for absblock, tabItemIcon, TabView components.

## Description

Updated new features in tabs component for tabsblock, tabItemIcon, TabView components.
This tabs component supports for Image, heading and embedding markdown documents.
Supported slots like image, heading, tabViewContent, theme, orientation, ..etc

## Related Issue
No related issues.

## Motivation and Context
Tabs wrapper feature is missing this adobe theme's, Existing tabs component couldn't able to markdown file. So that is why we updated this component.

## How Has This Been Tested?
We have done unit testing and still its in work in progress.

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/gatsby-theme-aio/src/components/Tabs/index.js [hunk 1]
```diff
@@ -10,10 +10,13 @@
  * governing permissions and limitations under the License.
  */
 
-import React, { useEffect, forwardRef } from 'react';
+import React, { useEffect, useState, useRef, createRef, forwardRef, cloneElement } from 'react';
 import classNames from 'classnames';
-import '@spectrum-css/tabs';
+import PropTypes from "prop-types";
```
Comment (icaraps) [inline packages/gatsby-theme-aio/src/components/Tabs/index.js:15]: Can you pls run `yarn format` to fix formatting issues like the double quotes here ?

File: packages/gatsby-theme-aio/src/components/Tabs/index.js [hunk 2]
```diff
@@ -24,52 +27,379 @@ const animateIndicator = (indicator, animate) => {
...
+  label,
+  ...props }, ref
+) => {
+  const Element = elementType;
+  return (
+    <Element
+      {...props}
+      ref={ref}
+      role="tab"
+      aria-selected={isSelected ? true : false}
```
Comment (icaraps) [inline packages/gatsby-theme-aio/src/components/Tabs/index.js:132]: `isSelected` is `bool` type so you can just write`isSelected` instead of checking the value.

File: packages/gatsby-theme-aio/src/components/Tabs/index.js [hunk 3]
```diff
@@ -24,52 +27,379 @@ const animateIndicator = (indicator, animate) => {
...
+      disabled={isDisabled ? true : false}
```
Comment (icaraps) [inline packages/gatsby-theme-aio/src/components/Tabs/index.js:133]: Do we need to support `isDisabled` ?

File: packages/gatsby-theme-aio/src/components/Tabs/index.js [hunk 4]
```diff
@@ -24,52 +27,379 @@ const animateIndicator = (indicator, animate) => {
...
     });
-  }, []);
+  }, [onFontsReady]);
 
   return (
     <div
       ref={ref}
       {...props}
-      className={classNames('spectrum-Tabs', 'spectrum-Tabs--horizontal', 'spectrum-Tabs--quiet', className)}
-      role="tablist">
```
Comment (icaraps) [inline packages/gatsby-theme-aio/src/components/Tabs/index.js:40]: Why did you remove `role="tablist"` especially since you're adding items with `role="tab"`?
`aria-orientation` should probably be added as well. It was missing from the beginning..

File: packages/gatsby-theme-aio/src/components/Tabs/index.js [hunk 5]
```diff
@@ -24,52 +27,379 @@ const animateIndicator = (indicator, animate) => {
...
+      {...props}
+      ref={ref}
+      role="tab"
+      aria-selected={isSelected ? true : false}
+      disabled={isDisabled ? true : false}
+      className={classNames(className, 'spectrum-Tabs-item', { 'is-selected': isSelected }, { 'is-disabled': isDisabled })}>
+      {icon ? <TabItemIcon icon={icon} isSelected={isSelected} isDisabled={isDisabled}></TabItemIcon> : null }
+      {label ? <Label> {label} </Label>: null}
+      {children}
+    </Element>
```
Comment (icaraps) [inline packages/gatsby-theme-aio/src/components/Tabs/index.js:138]: `aria-controls` should be added here as well and point to the content id.

File: packages/gatsby-theme-aio/src/components/Tabs/index.js [hunk 6]
```diff
@@ -24,52 +27,379 @@ const animateIndicator = (indicator, animate) => {
...
+                    .spectrum-Tabs-itemLabel{
+                      font-size:var(--spectrum-global-dimension-size-185);
+                      font-weight:bold;
+                      margin-left: var(--spectrum-global-dimension-size-200);
+                      line-height: initial;
+                      margin-top:var(--spectrum-global-dimension-size-100);
+                    }
+
+                    .spectrum-Icon{
+                      width: 40px;
```
Comment (icaraps) [inline packages/gatsby-theme-aio/src/components/Tabs/index.js:285]: Can you use `var(--spectrum-global-dimension-size-500)` instead ?

File: packages/gatsby-theme-aio/src/components/Tabs/index.js [hunk 7]
```diff
@@ -24,52 +27,379 @@ const animateIndicator = (indicator, animate) => {
...
+                      height: 40px;
+                    }
+
+                    @media only screen and (max-width: 480px) {
```
Comment (icaraps) [inline packages/gatsby-theme-aio/src/components/Tabs/index.js:289]: You're using `480px` multiple times, can you put it in a JS variable maybe in the same file at the top and reference it instead of hardcoding it ?

File: packages/gatsby-theme-aio/src/components/Tabs/index.js [hunk 8]
```diff
@@ -24,52 +27,379 @@ const animateIndicator = (indicator, animate) => {
...
+                    @media only screen and (device-width: ${MOBILE_SCREEN_WIDTH}) {
+                      max-width: ${layoutColumns(2)} !important;
+                    }
+                    &.is-disabled {
+                      pointer-events:none;
+                    }
+                    &:hover, &.is-selected {
+                        background-color: var(--spectrum-alias-background-color-gray-50, var(--spectrum-global-color-gray-50, var(--spectrum-semantic-gray-50-color-background)));
+                        color: var(--spectrum-global-color-gray-900);
+                        box-shadow: 0 3px 16px #46303029;
```
Comment (icaraps) [inline packages/gatsby-theme-aio/src/components/Tabs/index.js:311]: Is this a custom color or is it available as Spectrum CSS variable ?

File: packages/gatsby-theme-aio/src/components/Tabs/index.js [hunk 9]
```diff
@@ -24,52 +27,379 @@ const animateIndicator = (indicator, animate) => {
...
+        : null}
+      </div>
+    </section>
+  )
+}
+
+
+Tabs.propTypes = {
+  orientation: PropTypes.oneOf(["horizontal", "vertical"]),
+  density: PropTypes.oneOf(["regular", "compact"]),
```
Comment (icaraps) [inline packages/gatsby-theme-aio/src/components/Tabs/index.js:368]: Do we need to support multiple values for `density` ?

File: packages/gatsby-theme-aio/src/components/Tabs/index.js [hunk 10]
```diff
@@ -24,52 +27,379 @@ const animateIndicator = (indicator, animate) => {
...
+TabView.propTypes = {
+  isHidden: PropTypes.boolean,
+  className: PropTypes.string
+};
+
+TabsBlock.propTypes = {
+  theme: PropTypes.oneOf(["light", "dark", "lightest"]),
+  orientation: PropTypes.oneOf(["horizontal", "vertical"]),
+  heading: PropTypes.string,
+  tabViewContent: PropTypes.string,
```
Comment (icaraps) [inline packages/gatsby-theme-aio/src/components/Tabs/index.js:399]: Can you rename `tabViewContent` to just `content` to make the slot name more generic ? I can imagine other components that can make use of such a `content` slot.

**Review by icaraps — CHANGES_REQUESTED** [review]
Looking great ! Love the example with the tab content which is imported and referenced as slot !  

Just a few small comments and questions.

**Comment by icaraps** [comment]
Oh one more thing .. Can you pls update the README.md and mention the TabsBlock component ?

File: packages/gatsby-theme-aio/src/components/Tabs/index.js (hunk 1 above)
Comment (jsmitrah) [reply packages/gatsby-theme-aio/src/components/Tabs/index.js]: This is done.

File: packages/gatsby-theme-aio/src/components/Tabs/index.js (hunk 2 above)
Comment (jsmitrah) [reply packages/gatsby-theme-aio/src/components/Tabs/index.js]: This is done

File: packages/gatsby-theme-aio/src/components/Tabs/index.js (hunk 4 above)
Comment (jsmitrah) [reply packages/gatsby-theme-aio/src/components/Tabs/index.js:40]: Yes, you are right, This was removed by mistake I think. Now it's updated.

File: packages/gatsby-theme-aio/src/components/Tabs/index.js (hunk 5 above)
Comment (jsmitrah) [reply packages/gatsby-theme-aio/src/components/Tabs/index.js]: Added

File: packages/gatsby-theme-aio/src/components/Tabs/index.js (hunk 6 above)
Comment (jsmitrah) [reply packages/gatsby-theme-aio/src/components/Tabs/index.js]: Done.

File: packages/gatsby-theme-aio/src/components/Tabs/index.js (hunk 7 above)
Comment (jsmitrah) [reply packages/gatsby-theme-aio/src/components/Tabs/index.js]: This is done as you mentioned above

File: packages/gatsby-theme-aio/src/components/Tabs/index.js (hunk 8 above)
Comment (jsmitrah) [reply packages/gatsby-theme-aio/src/components/Tabs/index.js]: This Spectrum CSS variable.

File: packages/gatsby-theme-aio/src/components/Tabs/index.js (hunk 10 above)
Comment (jsmitrah) [reply packages/gatsby-theme-aio/src/components/Tabs/index.js]: This is done.

File: packages/gatsby-theme-aio/src/components/Tabs/index.js (hunk 3 above)
Comment (jsmitrah) [reply packages/gatsby-theme-aio/src/components/Tabs/index.js]: **isDisabled**  prop is removed.

File: packages/gatsby-theme-aio/src/components/Tabs/index.js (hunk 9 above)
Comment (jsmitrah) [reply packages/gatsby-theme-aio/src/components/Tabs/index.js]: **density**  prop is removed

**Review by icaraps — APPROVED** [review]
